python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,250 @@
+        doc="Intra-focal Donut Postage Stamp Images",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="StampsBase",
+        name="donutStampsIntra",
+    )
+    outputZernikesRaw = connectionTypes.Output(
+        doc="Zernike Coefficients from all donuts",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="NumpyArray",
+        name="zernikeEstimateRaw",
+    )
+    outputZernikesAvg = connectionTypes.Output(
+        doc="Zernike Coefficients averaged over donuts",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="NumpyArray",
+        name="zernikeEstimateAvg",
+    )
+    zernikes = connectionTypes.Output(
+        doc="Zernike Coefficients for individual donuts and average over donuts",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="AstropyQTable",
+        name="zernikes",
+    )
+    donutQualityTable = connectionTypes.Output(
+        doc="Quality information for donuts",
+        dimensions=("visit", "detector", "instrument"),
+        storageClass="AstropyQTable",
+        name="donutQualityTable",
+    )
+
+
+class CalcZernikesNeuralTaskConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=CalcZernikesNeuralTaskConnections,  # type: ignore
+):
+    """Configuration for CalcZernikesNeuralTask.
+
+    Attributes
+    ----------
+    wavenet_path : str
+        Model Weights Path for wavenet
+    alignet_path : str
+        Model Weights Path for alignet
+    aggregatornet_path : str
+        Model Weights Path for aggregatornet
+    dataset_param_path : str
+        datasetparam path
+    device : str
+        Device to use for calculations
+    """
+
+    wavenet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for wavenet",
+        dtype=str
+    )
+    alignet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for alignet",
+        dtype=str
+    )
+    aggregatornet_path = pexConfig.Field(  # type: ignore
+        doc="Model Weights Path for aggregatornet",
+        dtype=str
+    )
+    dataset_param_path = pexConfig.Field(  # type: ignore
+        doc="datasetparam path",
+        dtype=str
+    )
+    device = pexConfig.Field(  # type: ignore
+        doc="Device to use for calculations",
+        dtype=str,
+        default="cuda"
+    )
+
+
+class CalcZernikesNeuralTask(pipeBase.PipelineTask):
+    """Neural network-based Zernike estimation task using TARTS.
+
+    This class uses the TARTS (Telescope Active Optics Real-Time System)
+    neural network models to estimate Zernike coefficients from pairs of
+    intra and extra-focal exposures.
+    """
+    _DefaultName = "calcZernikesNeuralTask"
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+        # Define default Noll indices (zk terms 4-23, excl piston, tip, tilt)
+        self.nollIndices = np.arange(4, 24)
```

Comment:
what is this used for? I only see it used once to get the length of this. You probably need to remove it? In addition, you probably want to have this as a configuration, you don't want to hardcode it here